### PR TITLE
Insert particles

### PR DESCRIPTION
**Pull request by shahabgol**
A minor bug in calculation of IDs of new particles in multiple insertions is fixed.
The IDs of new particles started from 0 during each insertion. It is now changed to the number of particles in the system.

### CODE REVIEW

**Comment by bangerth** [comment]
@gassmoeller -- what do you say?

**Comment by blaisb** [comment]
> @gassmoeller -- what do you say?

I am not @gassmoeller, but I have a quick comment. Would it be better to use the next_free_particle_index() of the particle_handler class instead? The underlying assumption though is still that there is no "hole" in the particle numbering.

File: source/particles/particle_handler.cc [hunk 1]
```diff
@@ -521,7 +521,8 @@ namespace Particles
     // Calculate all starting points locally
     std::vector<unsigned int> particle_start_indices(n_mpi_processes);
```
Comment (luca-heltai) [inline source/particles/particle_handler.cc:522]:
```suggestion
    std::vector<types::particle_index> particle_start_indices(n_mpi_processes);
```

**Comment by luca-heltai** [comment]
Superseeded. Thanks for spotting this!

**Comment by bangerth** [comment]
@luca-heltai -- could you add what it is superseded by? Also, would it be worth adding the test anyway?

**Comment by luca-heltai** [comment]
Superseeded by #10510. These two were developed at the same time. The test is equivalent to one of the tests added in #10510.

**Pull request closed without merging**